src/pseudopeople/dataset.py

```diff
@@ -145,7 +145,7 @@ def _reformat_dates_for_noising(self) -> None:
                     # https://github.com/pandas-dev/pandas/issues/44764
                     # Year is already guaranteed to be 4-digit: https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#timeseries-timestamp-limits
                     is_na = data[column].isna()
-                    data_column = data.loc[~is_na, column]
+                    data_column = pd.to_datetime(data.loc[~is_na, column])
```

Comment:
What was the issue here?

Reply:
In certain cases this column was a string rather than a datetime object.

Reply:
I find it strange that you had to change this. Are you sure you're not inadvertently adding tests that change the type in an unrealistic way?

This entire for/if block is super confusing to me so it's hard for me to say. Why does line 142 include `and isinstance(column, str)`. Isn't `columns` just a column name and alwasy a str?

Reply:
Find which dataset(s) are/is causing this issue

Reply:
I think I wrote this code (in a different file, originally) and am happy to chat about it. The `isinstance` check was added [later for type checking purposes](https://github.com/ihmeuw/pseudopeople/commit/f23b752cbdad8f99d4781ea31b32508801679354#diff-4e6dfe005df0f993070b6b79753567ebc423aea3d835cf4f98362b79225c282eR142), which seems a bit suspect -- it seems like there should be a better way to make the typechecker know this is a string.